interop/tapir/src/main/scala/caliban/interop/tapir/WebSocketInterpreter.scala

```diff
@@ -0,0 +1,121 @@
+package caliban.interop.tapir
+
+import caliban._
+import caliban.interop.tapir.TapirAdapter._
+import caliban.interop.tapir.ws.Protocol
+import sttp.capabilities.zio.ZioStreams
+import sttp.model.{ headers => _ }
+import sttp.tapir.Codec.JsonCodec
+import sttp.tapir._
+import sttp.tapir.model.{ ServerRequest, UnsupportedWebSocketFrameException }
+import sttp.tapir.server.ServerEndpoint
+import sttp.ws.WebSocketFrame
+import zio._
+
+trait WebSocketInterpreter[-R, E] { self =>
```

Comment:
Should we seal it? Should users be able to make their own version of this?

Reply:
Ah, I forgot. Fixed.